gui/views/pattern_map/PatternMap.gd

```diff
@@ -729,17 +734,30 @@ func _get_pattern_at_position(at_position: Vector2) -> int:
 
 
 func _set_pattern_at_cursor(pattern_idx: int) -> void:
-	if not current_arrangement || not Controller.current_song:
-		return
+	_set_pattern_at_position(get_local_mouse_position(), pattern_idx)
 	
-	var cell := _get_cell_at_cursor()
+func _set_pattern_over_range(starting_pos: Vector2, ending_pos: Vector2, pattern_idx: int) -> void:
+	var _add_head_position = starting_pos
+	
+	# Jump to the next row; no point adding the pattern where it already is
```

Comment:
One last codestyle nitpick, but please put a period at the end of code comments. Also it should likely say "column", and not "row" :)